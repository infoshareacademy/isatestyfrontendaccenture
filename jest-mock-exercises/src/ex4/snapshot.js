export const createDivElement = (title) => {
    const div = document.createElement('div');
    div.innerHTML = title;
    return div;
}

export const createStyledParagraphElement = (text) => {
    const p = document.createElement('p');
    p.innerHTML = text;
    p.style = 'background-color: red; color: black;'
    return p;
}