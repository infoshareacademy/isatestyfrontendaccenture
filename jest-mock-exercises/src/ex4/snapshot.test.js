// Napisz test z uzyciem snapshota
import { createDivElement, createStyledParagraphElement } from './snapshot';

describe('createDivElement', () => {
    beforeEach(() => expect.hasAssertions());

    test('should create div element with title inline', () => {
        expect(createDivElement('title')).toMatchInlineSnapshot(`
      <div>
        title
      </div>
    `);
    });

    test('should create div element with title', () => {
        expect(createDivElement('title')).toMatchSnapshot();
    });
});

describe('createStyledParagraphElement', () => {
    beforeEach(() => expect.hasAssertions());

    test('should create p element with title and proper styles inline', () => {
        expect(createStyledParagraphElement('title')).toMatchInlineSnapshot(`
      <p
        style="background-color: red; color: black;"
      >
        title
      </p>
    `);
    });
    test('should create p element with title and proper styles', () => {
        expect(createStyledParagraphElement('title')).toMatchSnapshot();
    });
});
