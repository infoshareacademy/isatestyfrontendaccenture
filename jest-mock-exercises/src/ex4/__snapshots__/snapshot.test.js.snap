// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`createDivElement should create div element with title 1`] = `
<div>
  title
</div>
`;

exports[`createStyledParagraphElement should create p element with title and proper styles 1`] = `
<p
  style="background-color: red; color: black;"
>
  title
</p>
`;
